protoc-gen-kotlin/lib/src/commonMain/kotlin/pbandk/gen/CodeGenerator.kt

```diff
@@ -121,37 +165,117 @@ open class CodeGenerator(
-                            line("messageDescriptor = this@Companion::descriptor,")
-                            line("name = \"${field.name}\",")
-                            line("number = ${field.number},")
-                            line("type = ${field.fieldDescriptorType(oneof != null)},")
-                            oneof?.let { line("oneofMember = true,") }
-                            field.jsonName?.let { line("jsonName = \"$it\",") }
-                            line("value = $fullTypeName::${field.kotlinFieldName}")
-                        }.line(")")
-                    }.line(")")
+            // val allFields = type.sortedStandardFieldsWithOneOfs()
+            if (needToChunk) {
+                line("val fieldsList = ArrayList<pbandk.FieldDescriptor<$fullTypeName, *>>(${chunkSize})")
+                allFields.chunked(chunkSize).forEachIndexed { index, chunk ->
+                    line("addFields${index}(fieldsList)")
                 }
-            }.line("}")
+            } else {
+                addFields(allFields, fullTypeName, line("val fieldsList = ArrayList<pbandk.FieldDescriptor<$fullTypeName, *>>(${allFields.size}).apply {"))
+            }
+
             line("pbandk.MessageDescriptor(").indented {
                 line("messageClass = $fullTypeName::class,")
                 line("messageCompanion = this,")
                 line("fields = fieldsList")
             }.line(")")
         }.line("}")
+
+        if (needToChunk) {
+            allFields.chunked(chunkSize).forEachIndexed { index, chunk ->
+                line("fun addFields${index}(fieldsList: ArrayList<pbandk.FieldDescriptor<$fullTypeName, *>>) {").indented {
+                    addFields(chunk, fullTypeName, line("fieldsList.apply {"))
+
+                }.line("}")
+            }
+        }
+    }
+
+    private fun addFields(
+        chunk: List<Pair<File.Field.Numbered, File.Field.OneOf?>>,
+        fullTypeName: String,
+        beginningLine: CodeGenerator
+    ): CodeGenerator {
+        return beginningLine.indented {
+            chunk.forEach { (field, oneof) ->
+                line("add(").indented {
+                    line("pbandk.FieldDescriptor(").indented {
+                        line("messageDescriptor = this@Companion::descriptor,")
+                        line("name = \"${field.name}\",")
+                        line("number = ${field.number},")
+                        line("type = ${field.fieldDescriptorType(oneof != null)},")
+                        oneof?.let { line("oneofMember = true,") }
+                        field.jsonName?.let { line("jsonName = \"$it\",") }
+                        field.options?.let { generateFieldOptions(it) }
+                        line("value = $fullTypeName::${field.kotlinFieldName}")
+                    }.line(")")
+                }.line(")")
+            }
+        }.line("}")
+    }
+
+    private fun generateFieldOptions(fieldOptions: FieldOptions) {
```

Comment:
Looking for suggestions here. Field options has 2 supported options which are both optional. This logic is really only trying to accomplish whether to add a trailing `,` or not. In my mind it has to look forward to know if it needs to  add one or not.

Reply:
This looked like it'd be such a simple problem, but every time I'd try to write up a suggested solution I'd run into a problem (I'm sure the same problems that led you to your current version). What I ultimately came up with is a bit more brute-force, but I think it ends up being more readable:

```kt
    private fun generateFieldOptions(fieldOptions: FieldOptions) {
        if (fieldOptions != FieldOptions.defaultInstance) {
            line("options = pbandk.wkt.FieldOptions(").indented {
                fieldOptions.deprecated?.let {
                    lineBegin("deprecated = $it")
                }
                if (fieldOptions.unknownFields.isEmpty()) {
                    lineEnd()
                } else {
                    lineEnd(",")
                    generateUnknownFields(fieldOptions.unknownFields)
                }
            }.line("),")
        }
    }
 ```